cmd/deployment/create.go

```diff
@@ -190,25 +196,32 @@ func init() {
 	createCmd.Flags().Int32("appsearch-zones", 1, "Number of zones the App Search instances will span")
 	createCmd.Flags().Int32("appsearch-size", 2048, "Memory (RAM) in MB that each of the App Search instances will have")
 
-	// The following flags will remain hidden until reads for deployment templates are available on ESS
-	createCmd.Flags().MarkHidden("deployment-template")
-	createCmd.Flags().MarkHidden("version")
-	createCmd.Flags().MarkHidden("name")
-	createCmd.Flags().MarkHidden("generate-payload")
-	createCmd.Flags().MarkHidden("es-ref-id")
-	createCmd.Flags().MarkHidden("es-zones")
-	createCmd.Flags().MarkHidden("es-size")
-	createCmd.Flags().MarkHidden("topology-element")
-	createCmd.Flags().MarkHidden("plugin")
-	createCmd.Flags().MarkHidden("kibana-ref-id")
-	createCmd.Flags().MarkHidden("kibana-zones")
-	createCmd.Flags().MarkHidden("kibana-size")
-	createCmd.Flags().MarkHidden("apm")
-	createCmd.Flags().MarkHidden("apm-ref-id")
-	createCmd.Flags().MarkHidden("apm-zones")
-	createCmd.Flags().MarkHidden("apm-size")
-	createCmd.Flags().MarkHidden("appsearch")
-	createCmd.Flags().MarkHidden("appsearch-ref-id")
-	createCmd.Flags().MarkHidden("appsearch-zones")
-	createCmd.Flags().MarkHidden("appsearch-size")
+	createCmd.Flags().Bool("enterprise-search", false, "Enables Enterprise Search for the deployment")
+	createCmd.Flags().String("enterprise-search-ref-id", "main-enterprise_search", "Optional RefId for the Enterprise Search deployment")
+	createCmd.Flags().Int32("enterprise-search-zones", 1, "Number of zones the Enterprise Search instances will span")
+	createCmd.Flags().Int32("enterprise-search-size", 4096, "Memory (RAM) in MB that each of the Enterprise Search instances will have")
+}
+
+func setDefaultTemplate(region string) string {
+	if strings.Contains(region, "azure") {
+		region = "azure"
+	}
+
+	if strings.Contains(region, "gcp") {
+		region = "gcp"
+	}
+
+	var dt string
+	switch region {
+	case "azure":
+		dt = "azure-io-optimized"
+	case "gcp":
+		dt = "gcp-io-optimized"
+	case "ece-region":
+		dt = "default"
+	default:
+		dt = "aws-io-optimized-v2"
+	}
+
+	return dt
```

Comment:
I would have used early returns since it seems more declarative than assigning vars, but it doesn't really matter.

```suggestion
	switch region {
	case "azure":
		return "azure-io-optimized"
	case "gcp":
		return "gcp-io-optimized"
	case "ece-region":
		return "default"
	default:
		return "aws-io-optimized-v2"
	}
```